be/src/vec/core/block.cpp

```diff
@@ -872,6 +889,10 @@ void MutableBlock::add_rows(const Block* block, const int* row_begin, const int*
 }
 
 void MutableBlock::add_rows(const Block* block, size_t row_begin, size_t length) {
+    if (_type == BlockType::DYNAMIC) {
```

Comment:
It may be more efficient to add UNLIKELY since most tables are without dynamic schema.